Tests/ResultTests.swift

```diff
@@ -277,4 +280,194 @@ class ResultTestCase: BaseTestCase {
             XCTFail("flatMap should preserve the failure error")
         }
     }
+    
+    // MARK: - map/flatMapError Tests
+    
+    func testMapErrorTransformsErrorValue() {
+        // Given
+        struct ResultError: Error {}
+        struct OtherError: Error { let error: Error }
+        let result: Result<String> = .failure(ResultError())
+        
+        // When
+        let mappedResult = result.mapError { OtherError(error: $0) }
+        
+        // Then
+        if let error = mappedResult.error {
+            XCTAssertTrue(error is OtherError)
+        } else {
+            XCTFail("mapError should transform error value")
+        }
+    }
+    
+    func testMapErrorPreservesSuccessError() {
+        // Given
+        struct ResultError: Error {}
+        struct OtherError: Error { let error: Error }
+        let result: Result<String> = .success("success")
+        
+        // When
+        let mappedResult = result.mapError { OtherError(error: $0) }
+        
+        // Then
+        XCTAssertEqual(mappedResult.value, "success")
+    }
+    
+    func testFlatMapErrorTransformsErrorValue() {
+        // Given
+        struct ResultError: Error {}
+        struct OtherError: Error { let error: Error }
+        let result: Result<String> = .failure(ResultError())
+        
+        // When
+        let mappedResult = result.flatMapError { OtherError(error: $0) }
+        
+        // Then
+        if let error = mappedResult.error {
+            XCTAssertTrue(error is OtherError)
+        } else {
+            XCTFail("mapError should transform error value")
+        }
+    }
+    
+    func testFlatMapErrorCapturesThrownError() {
+        // Given
+        struct ResultError: Error {}
+        struct OtherError: Error {
+            let error: Error
+            init(error: Error) throws { throw ThrownError() }
+        }
+        struct ThrownError: Error {}
+        let result: Result<String> = .failure(ResultError())
+        
+        // When
+        let mappedResult = result.flatMapError { try OtherError(error: $0) }
```

Comment:
Addressed in [9bf29](https://github.com/Alamofire/Alamofire/pull/2135/commits/131354b48086ea94f3b3d9303347d2f64079bf29).